### PR TITLE
Develop pull requests

### PR DESCRIPTION
**Pull request by daylightstudio**
This pull requests adjusts 2 things. One is an issue when the 2nd and 3rd parameters have null passed to them it throws an error. The second adjusts the field meta data function (_fieldData) to provide an array of options handy for enum field types or even float.

### CODE REVIEW

File: system/HTTP/Response.php [hunk 1]
```diff
@@ -559,7 +559,7 @@ public function sendBody()
 	 * @return $this
 	 * @throws \CodeIgniter\HTTP\RedirectException
 	 */
-	public function redirect(string $uri, string $method = 'auto', int $code = null)
+	public function redirect(string $uri, $method = 'auto', $code = null)
```
Comment (lonnieezell) [inline system/HTTP/Response.php]: Removing the typehints here won't affect anything, as long as there is a default value (even null) it should accept null values just fine. Can you provide example usage that's causing the problem and verify that you're a) on the latest develop commit and b) running under PHP 7+.

File: system/Database/MySQLi/Connection.php [hunk 2]
```diff
@@ -416,6 +416,16 @@ public function _fieldData(string $table)
 			sscanf($query[$i]->Type, '%[a-z](%d)', $retval[$i]->type, $retval[$i]->max_length
 			);
 
+			preg_match('#[a-z]+\((\'.+\')\)#', $query[$i]->Type, $matches);
```
Comment (lonnieezell) [inline system/Database/MySQLi/Connection.php]: This block needs documentation and additional tests written for it.

File: system/HTTP/Response.php (hunk 1 above)
Comment (daylightstudio) [reply system/HTTP/Response.php]: For the redirect issue, I was simply running the following in my controller after a post request:
```
redirect()->back()->withInput();
```

And getting the following error:
```
TypeError
Argument 2 passed to CodeIgniter\HTTP\Response::redirect() must be of the type string, null given, called in /Library/WebServer/Documents/daylight/FUEL/v2.0_CI/system/HTTP/RedirectResponse.php on line 105	
```
I'm running PHP 7.1.8

File: system/Database/MySQLi/Connection.php (hunk 2 above)
Comment (daylightstudio) [reply system/Database/MySQLi/Connection.php]: For enum values (and float) field types, their options are stored in the type property and need to be extracted out. The regular expression grabs everything in the parenthesis, creates an array of properties split with ',' and if it's no empty it creates the options parameter containing the enum options. Are there any tests for the getFieldData method other then in the ForgeTest?

Comment (lonnieezell) [reply system/Database/MySQLi/Connection.php]: I don't believe there are any other tests for that method other than ForgeTest.

File: system/HTTP/Response.php (hunk 1 above)
Comment (lonnieezell) [reply system/HTTP/Response.php]: It looks like the deal is that the `previous_url()` method in the URL helper is returning null. So the solution here doesn't appear to be removing the typehint, but adding a default null value and ensuring things work from there. 

```
redirect(string $uri=null, $method='auto', $code=null);
```

**Comment by jim-parry** [comment]
Being picky, but we expect commits to be GPG-signed :)

**Pull request closed without merging**